src/desdemona/query.clj

```diff
@@ -0,0 +1,25 @@
+(ns desdemona.query
+  (:require [clojure.core.logic :as l]))
+
+(defn ^:private gen-query
+  "Expands a query and events to a core.logic program that executes
+  it."
+  [n-answers query events]
+  `(l/run ~n-answers [results#]
+     (l/fresh [~'x]
+       (l/== [~'x] results#)
+       (l/membero ~'x ~events)
+       ~query)))
```

Comment:
Yea. My goal is just to have a comment that makes this a bit easier to parse for people who (a) aren't familiar with core.logic and (b) don't understand why some of the syntactic gymnastics are necessary :stuck_out_tongue_winking_eye:.